dds/DCPS/DiscoveryBase.h

```diff
@@ -128,6 +178,12 @@ namespace OpenDDS {
       bool reader_done_, writer_done_;
       ACE_Thread_Mutex mtx_;
       ACE_Condition_Thread_Mutex cnd_;
+
+      // thread reporting
+      TimeDuration interval;
+      ThreadStatus* status;
+      ACE_thread_t tid;
+      bool has_timeout;
```

Comment:
Why have a seperate bool as member that indicates interval>0, use inline helper method? When you keep it, interval and has_timeout could be const when you initialise them directly